src/tasks/LargeFileUploadTask.ts

```diff
@@ -221,37 +237,51 @@ export class LargeFileUploadTask {
-				err.name = "Invalid Session";
-				throw err;
-			}
-			const fileSlice = this.sliceFile(nextRange);
-			const rawResponse = await this.uploadSliceGetRawResponse(fileSlice, nextRange, this.file.size);
-			if (!rawResponse) {
-				throw new GraphClientError("Something went wrong! Large file upload slice response is null.");
-			}
+		const progressCallBack = this.options.progressCallBack;
+		try {
+			while (!this.uploadSession.isCancelled) {
+				const nextRange = this.getNextRange();
+				if (nextRange.maxValue === -1) {
+					const err = new Error("Task with which you are trying to upload is already completed, Please check for your uploaded file");
+					err.name = "Invalid Session";
+					throw err;
+				}
+				const fileSlice = await this.file.sliceFile(nextRange);
+				const rawResponse = await this.uploadSliceGetRawResponse(fileSlice, nextRange, this.file.size);
+				if (!rawResponse) {
+					throw new GraphClientError("Something went wrong! Large file upload slice response is null.");
+				}
 
-			const responseBody = await GraphResponseHandler.getResponse(rawResponse);
-			/**
-			 * (rawResponse.status === 201) -> This condition is applicable for OneDrive, PrintDocument and Outlook APIs.
-			 * (rawResponse.status === 200 && responseBody.id) -> This additional condition is applicable only for OneDrive API.
-			 */
-			if (rawResponse.status === 201 || (rawResponse.status === 200 && responseBody.id)) {
-				return UploadResult.CreateUploadResult(responseBody, rawResponse.headers);
-			}
-
-			/* Handling the API issue where the case of Outlook upload response property -'nextExpectedRanges'  is not uniform.
-			 * https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/39
-			 */
-			const res: UploadStatusResponse = {
-				expirationDateTime: responseBody.expirationDateTime,
-				nextExpectedRanges: responseBody.NextExpectedRanges || responseBody.nextExpectedRanges,
-			};
+				const responseBody = await GraphResponseHandler.getResponse(rawResponse);
+				/**
+				 * (rawResponse.status === 201) -> This condition is applicable for OneDrive, PrintDocument and Outlook APIs.
+				 * (rawResponse.status === 200 && responseBody.id) -> This additional condition is applicable only for OneDrive API.
+				 */
+				if (rawResponse.status === 201 || (rawResponse.status === 200 && responseBody.id)) {
+					const uploadResult = UploadResult.CreateUploadResult(responseBody, rawResponse.headers);
+					if (progressCallBack && progressCallBack.completed) {
+						progressCallBack.completed(uploadResult, progressCallBack.extraCallbackParams);
+					}
+					return uploadResult;
+				}
 
-			this.updateTaskStatus(res);
+				/* Handling the API issue where the case of Outlook upload response property -'nextExpectedRanges'  is not uniform.
+				 * https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/39
+				 */
+				const res: UploadStatusResponse = {
+					expirationDateTime: responseBody.expirationDateTime,
```

Comment:
the casing for that property differs as well from one workload to another

Reply:
changed to `expirationDateTime: responseBody.expirationDateTime || responseBody.ExpirationDateTime`